### PR TITLE
GraalVM support

### PR DESCRIPTION
**Pull request by cap10morgan**
This PR adds basic graalvm native-image support to ledger. Part of that was porting it to http-kit (from aleph), so that's in here.

The tests pass, it builds, it runs, it transacts, it queries. That being said, I'd be very surprised if there weren't still bugs in here that aren't in the JVM version (I am hoping that this doesn't introduce any new bugs into the JVM version, though). It's a start!

Next I'm going to get the native-image building in Docker, document how to do it both there and outside Docker, test clustering, etc.

### CODE REVIEW

**Review by zonotope — APPROVED** [review]
👍🏾

File: src/fluree/db/ledger/transact/core.clj [hunk 1]
```diff
@@ -183,24 +185,24 @@
               (nil? existing-flake) (recur r)
 
               ;; lookup subject matches subject, will end up ignoring insert downstream unless :retractDuplicates is true
-              (= (.-s ^Flake existing-flake) _id) (recur r)
+              (= (:s existing-flake) _id) (recur r)
```
[the remark this answers is not in the document]
Comment (cap10morgan) [reply src/fluree/db/ledger/transact/core.clj:188]: The fastest is `.-attr` w/ type hint, but the keyword access isn't that much slower so outside of perf-critical code it's probably OK.

They're both wayyyyyyyyyyyyyyyyyyyy faster than reflection-causing `.-attr` without a type hint or a go-macro mangled one.

**Comment by cap10morgan** [comment]
Note that merging this is blocked by a few things currently.

**Review by ldw1007 — APPROVED** [review]
lgtm

Created a task, FC-1247, to review all handlers and decide whether they should be async or sync processes.

**Pull request merged**